internal/util/validation/quadlet.go

```diff
@@ -78,8 +88,14 @@ func ValidateQuadletSpec(spec *common.QuadletReferences, path string) []error {
 		if spec.Image == nil {
 			errs = append(errs, fmt.Errorf(".image quadlet must have an Image key"))
 		} else {
-			if err := ValidateOciImageReferenceStrict(spec.Image, "image.image"); err != nil {
-				errs = append(errs, err...)
+			if fleetTemplate {
+				if err := ValidateOciImageReferenceWithTemplates(spec.Image, "image.image"); err != nil {
+					errs = append(errs, err...)
+				}
+			} else {
+				if err := ValidateOciImageReferenceStrict(spec.Image, "image.image"); err != nil {
+					errs = append(errs, err...)
+				}
```

Comment:
nit: seems like we could have a ValidateOciImageReference(spec.Image, "image.image", fleetTemplate) that eliminates these else conditions